### DISCUSSION TITLE
"window is not defined" when using JsSIP in Node.JS

### DISCUSSION THREAD

**Discussion by FloppiTuna** [versatica/JsSIP#864] [Q&A]
I'm able to create a connection between JsSIP and my Asterisk/FreePBX server, however as soon as it receives/sends a call, it throws the following exception:

```
/Users/tuna/src/Quizmaster/node_modules/jssip/lib-es5/RTCSession.js:269
      if (!window.RTCPeerConnection) {
      ^

ReferenceError: window is not defined
    at RTCSession.connect (/Users/tuna/src/Quizmaster/node_modules/jssip/lib-es5/RTCSession.js:269:7)
    at UA.call (/Users/tuna/src/Quizmaster/node_modules/jssip/lib-es5/UA.js:241:15)
    at file:///Users/tuna/src/Quizmaster/index.js:36:18
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:337:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:106:12)
```

The code:
```
import JsSIP from 'jssip';

var socket = new JsSIP.WebSocketInterface('ws://x.x.x.x:8088/ws');
var configuration = {
  sockets  : [ socket ],
  uri      : 'sip:xxxx@x.x.x.x',
  password : '....'
};

var ua = new JsSIP.UA(configuration);

ua.start();

ua.on('registered', () => {
    console.log('Registered!');
})

ua.on('newRTCSession', function(data){
   console.log("Call!");
});
```

**Comment by Subhan00** [comment]
Hi @FloppiTuna, JsSIP is a browser-based SIP library designed to run on the client side, not directly in Node.js. If you’re trying to use it, here’s a quick guide to set it up properly:

Create an HTML page.

In the <head> section, include the JsSIP library like this: <script src="./libs/jssip-3.10.0.min.js"></script>

In the <body>, link a custom JavaScript file where you’ll write your SIP logic.

Then in that JS file, you can initialize and register the UA like this:

var socket = new JsSIP.WebSocketInterface('ws://x.x.x.x:8088/ws');
var configuration = {
  sockets  : [ socket ],
  uri      : 'sip:xxxx@x.x.x.x',
  password : 'your_password_here'
};

var ua = new JsSIP.UA(configuration);

ua.start();

ua.on('registered', () => {
    console.log('Registered!');
});

ua.on('newRTCSession', function(data) {
    console.log('Incoming or outgoing call detected');
});

If you face any issues while setting it up, feel free to ask — happy to help!
